### PR TITLE
[Graph Partition] add symints to get_graph_inputs

### PR DESCRIPTION
**Pull request by BoyuanFeng**
During `codegen_inputs`, we check whether there are undefined symbols:
https://github.com/pytorch/pytorch/blob/65b1aedd09e98fcafcdd893ca4924f4fa598fd18/torch/_inductor/codegen/wrapper.py#L1668-L1674

Previously, for graph partition inputs, we do not explicitly add symints.
https://github.com/pytorch/pytorch/blob/65b1aedd09e98fcafcdd893ca4924f4fa598fd18/torch/_inductor/codegen/wrapper.py#L3265-L3272
We relied on sizes/strides of TensorBox for codegen symint inputs.  For example, a tensor with shape `[s0, 2]` will implicitly codegen `s0` as an input here. This works fine in most cases since backed symint has to come from some tensor shapes.
https://github.com/pytorch/pytorch/blob/65b1aedd09e98fcafcdd893ca4924f4fa598fd18/torch/_inductor/codegen/wrapper.py#L1624-L1632

In rare cases, this does not work. One example is saved tensors for backward where a tensor may have shape `[2*s0, 2]`. Since `2*s0` is an expression but not a symbol, `codegen_input_symbol_assignment` would not handle `s0` and later there would be an error when `_verify_input_symbol_assignment`.   

The fix is add symints to `get_graph_inputs`. An alternative way is to update `codegen_input_symbol_assignment` but I want to minimize the change to graph partition only.

cc @H-Huang @awgu @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -14824,6 +14824,27 @@ def f(x, y):
...
+            def f(x, w):
+                y = torch.cat((x, x), dim=0)
+                z = y @ w
+                return z @ z.T
+
+            compiled_f = torch.compile(f)
+
+            for shape in (2, 3):
+                torch.manual_seed(42)
+                eager_x = torch.randn(shape, 2, device="cuda")
```
Comment (etaf) [inline test/inductor/test_torchinductor.py:14838]: Hi, May I suggest to replace the hard code `"cuda"` with `GPU_TYPE` in this case ? Otherwise the `"cuda"` will fail on other GPUs.

File: torch/_inductor/scheduler.py [hunk 2]
```diff
@@ -4249,7 +4249,15 @@ def filter_symbols(
             *(get_input_node_symbols(node) for _, node in input_nodes.items())
         )
 
-        return filter_symbols(candidate_symbols)
+        candidate_symbols = filter_symbols(candidate_symbols)
+
+        res: OrderedSet[sympy.Symbol] = OrderedSet()
+        for s in candidate_symbols:
+            symplified_s = V.graph.sizevars.simplify(s)
```
Comment (eellison) [inline torch/_inductor/scheduler.py:4258]: can you say why we simplify the symbol here ? Is it possible that we would have an expr like `s0 + s1 * 0` that when simplified would remove the s1 symbol, and then we would expect to need it later ?

File: torch/_inductor/scheduler.py [hunk 3]
```diff
@@ -4249,7 +4249,15 @@ def filter_symbols(
...
+            # use free_symbols only when s is simplified to an Integer or expr
+            res.update(symplified_s.free_symbols)
```
Comment (eellison) [inline torch/_inductor/scheduler.py:4260]: we should sort the free symbols like with https://github.com/pytorch/pytorch/pull/154846

cc @rec

**Review by eellison — APPROVED** [review]
looks good!

File: torch/_inductor/scheduler.py (hunk 2 above)
Comment (BoyuanFeng) [reply torch/_inductor/scheduler.py:4258]: We simplify the symbol to match existing codegen behavior:
https://github.com/pytorch/pytorch/blob/a4da1d4a471ccfc3342c7b8ade95bc174c263144/torch/_inductor/codegen/wrapper.py#L1714-L1715

Candidate symbol s can only be a symbol since it is the output of `filter_symbols`. `s` cannot be an expr. The `simplified_s` could be a symbol, expr, or integer. So we use `symplified_s.free_symbols` as the result.

File: torch/_inductor/scheduler.py [hunk 4]
```diff
@@ -4259,7 +4259,7 @@ def filter_symbols(
             # use free_symbols only when s is simplified to an Integer or expr
             res.update(symplified_s.free_symbols)
 
-        return res
+        return OrderedSet(sorted(res))
```
Comment (eellison) [inline torch/_inductor/scheduler.py]: I think you need to use the attrgetter. see:

```
sorted((x + y).free_symbols)
```

I meant to give full repro here haha ^^ but anyway that errors
